src/libs/Middleware/Retry.js

```diff
@@ -18,11 +18,24 @@ function Retry(response, request, isFromSequentialQueue) {
 
             if (isFromSequentialQueue) {
                 const retryCount = PersistedRequests.incrementRetries(request);
-                Log.info('Persisted request failed', false, {retryCount, command: request.command, error: error.message});
-                if (retryCount >= CONST.NETWORK.MAX_REQUEST_RETRIES) {
-                    Log.info('Request failed too many times, removing from storage', false, {retryCount, command: request.command, error: error.message});
-                    PersistedRequests.remove(request);
+                const logParams = {retryCount, command: request.command, error: error.message};
+                Log.info('Persisted request failed', false, logParams);
+                let logMessage = '';
+
+                if (error.message === CONST.ERROR.EXPENSIFY_SERVICE_INTERRUPTED) {
+                    logMessage = 'Request failed because the Expensify backend is not available';
+                } else if (error.message === CONST.ERROR.THROTTLED) {
+                    logMessage = 'Request is being throttled';
+                } else if (retryCount >= CONST.NETWORK.MAX_REQUEST_RETRIES) {
+                    logMessage = 'Request failed too many times';
+                } else {
+                    // We are ok to do nothing and allow the SequentialQueue to retry the request
+                    return;
                 }
+
+                // We shouldn't retry the request so log a message explaining why and remove the request from storage.
+                Log.info(`${logMessage}, removing from storage`, false, logParams);
+                PersistedRequests.remove(request);
```

Comment:
hmmmm if there are several requests that are dependent, then discarding one will make the following ones fail too, right?